### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by TerrorJack**
* Repo secrets updated, `NPM_TOKEN` should work now, also added main repo token
* The protobuf autogen JS files are updated according to latest definition
* All CI pipelines now fetch from `public/ic` GitLab repo if necessary
* `integration-test` no longer perform any rust building, it simply fetches latest `master` branch blessed artifacts and use those, so the pipeline should be much faster
* We now prefer node-14 over 16 whenever possible, rationale: we are sticking to version 1 of `package-lock.json` for now, node-14 is the last LTS version that ships with npm defaulting to version 1
* Updated `package-lock.json`, drop redundant patch

### CODE REVIEW

File: .github/workflows/integration-test.yml [hunk 1]
```diff
@@ -97,61 +72,60 @@ jobs:
...
-          restore-keys: 1-cargo-cache-
-          path: |
-            ~/.cargo
-            dfinity/rs/target
+      - name: setup-nix
+        uses: cachix/install-nix-action@v16
+
+      - name: setup-nghttpx
+        run: |
+          nix profile install nixpkgs#nghttp2
```
Comment (roman-kashitsyn) [inline .github/workflows/integration-test.yml:82]: I believe there is a `nghttp2` package in the ubuntu-latest that we use for the build. Is there any benefit in installing another package management system?

File: .github/workflows/protobuf-check.yml [hunk 2]
```diff
@@ -10,31 +10,30 @@ jobs:
...
-        with:
-          repository: dfinity-lab/dfinity
-          token: ${{ secrets.GH_TOKEN }}
-          path: dfinity
+      - name: checkout-ic
+        run: |
+          pushd ${{ runner.temp }}
+          curl --header "PRIVATE-TOKEN: ${{ secrets.GITLAB_TOKEN }}" https://gitlab.com/dfinity-lab/public/ic/-/archive/master/ic-master.tar.gz | tar xz
+          mv ic-master ic
+          popd
```
Comment (roman-kashitsyn) [inline .github/workflows/protobuf-check.yml:25]: We can get the sources from https://github.com/dfinity/ic, it's the public mirror of https://gitlab.com/dfinity-lab/public/ic. This way we don't need a token.

**Review by roman-kashitsyn — APPROVED** [review]

File: .github/workflows/integration-test.yml (hunk 1 above)
Comment (TerrorJack) [reply .github/workflows/integration-test.yml:82]: `nghttp2` in ubuntu doesn't work. I've yet to figure out why, but the nix version is sufficient to keep stuff working for now.

File: .github/workflows/protobuf-check.yml (hunk 2 above)
Comment (TerrorJack) [reply .github/workflows/protobuf-check.yml:25]: Ah, nice point! Will pull from public mirror instead.

**Pull request merged**